src/application/clients/crux-wallet-client.ts

```diff
@@ -0,0 +1,211 @@
+            const assetIdAddressMap = this._cruxUser.getAddressMap();
+            return this._getCruxAssetTranslator().assetIdAddressMapToSymbolAddressMap(assetIdAddressMap);
+        }
+        return {};
+    }
+
+    @throwCruxClientError
+    public putAddressMap = async (newAddressMap: IAddressMapping): Promise<{success: IPutAddressMapSuccess, failures: IPutAddressMapFailures}> => {
+        await this._initPromise;
+        if (!this._keyManager) {
+            throw ErrorHelper.getPackageError(null, PackageErrorCode.PrivateKeyRequired);
+        }
+        const {assetAddressMap, success, failures} = this._getCruxAssetTranslator().symbolAddressMapToAssetIdAddressMap(newAddressMap);
+        if (!this._cruxUser) {
+            throw ErrorHelper.getPackageError(null, PackageErrorCode.UserDoesNotExist);
+        }
+        const cruxUser = cloneValue(this._cruxUser);
+        cruxUser.addressMap = assetAddressMap;
+        this._cruxUser = await this._cruxUserRepository.save(cruxUser, this._keyManager);
+        return {success, failures};
+    }
+
+    @throwCruxClientError
+    public isCruxIDAvailable = async (cruxIDSubdomain: string): Promise<boolean> => {
+        await this._initPromise;
+        const cruxIdInput: InputIDComponents = {
+            domain: this.walletClientName,
+            subdomain: cruxIDSubdomain,
+        };
+        const cruxId = new CruxId(cruxIdInput);
+        return await this._cruxUserRepository.find(cruxId);
+    }
+
+    @throwCruxClientError
+    public registerCruxID = async (cruxIDSubdomain: string): Promise<void> => {
+        await this._initPromise;
+        if (!this._keyManager) {
+            throw ErrorHelper.getPackageError(null, PackageErrorCode.PrivateKeyRequired);
+        }
+        if (this._cruxUser) {
+            throw ErrorHelper.getPackageError(null, PackageErrorCode.ExistingCruxIDFound, this._cruxUser.cruxID);
+        }
+        const cruxIdInput: InputIDComponents = {
+            domain: this.walletClientName,
+            subdomain: cruxIDSubdomain,
+        };
+        const cruxId = new CruxId(cruxIdInput);
+        this._cruxUser = await this._cruxUserRepository.create(cruxId, this._keyManager);
+    }
+
+    @throwCruxClientError
+    public getAssetMap = async (): Promise<IResolvedClientAssetMap> => {
+        await this._initPromise;
+        if (this.resolvedClientAssetMapping) {
+            return this.resolvedClientAssetMapping;
+        }
+        return this._getCruxAssetTranslator().assetIdAssetMapToSymbolAssetMap(this.getCruxDomain().config.assetList);
+    }
+
+    private _getCruxUserByID = async (cruxIdString: string, tag?: string): Promise<CruxUser|undefined> => {
+        const cruxId = CruxId.fromString(cruxIdString);
+        return await this._cruxUserRepository.getByCruxId(cruxId, tag);
+    }
+
+    private _getCruxAssetTranslator = () => {
+        if (!this._cruxAssetTranslator) {
+            throw new BaseError(null, "");
```

Comment:
throw a meaningful error string at the least

Reply:
Updated, throwing packageError instead